src/Schema/Domain/Content/Worker/FieldDefinition/AddFieldDefinitionToDomainContentMutation.php

```diff
@@ -62,45 +65,54 @@ public function canWork(Builder $schema, array $args): bool
             && $args['ContentType'] instanceof ContentType
             && isset($args['FieldDefinition'])
             && $args['FieldDefinition'] instanceof FieldDefinition
-            && !$schema->hasTypeWithField($this->getCreateInputName($args['ContentType']), $this->getFieldDefinitionField($args['FieldDefinition']));
+            && !$schema->hasTypeWithField($this->nameCreateInputType($args), $this->nameFieldDefinitionField($args));
     }
 
     /**
-     * @param ContentType $contentType
+     * @param array $args
      *
      * @return string
      */
-    protected function getCreateInputName(ContentType $contentType): string
+    protected function nameCreateInputType(array $args): string
```

Comment:
I still don't follow why you prefer passing array of not well-known objects instead of strictly-typed `ContentType` for this method. Am I missing the bigger picture here?

Reply:
Because these methods are called over and over in the worker classes. It increases readability a lot. I actually started with explicit arguments, but decided to change it when I saw it grow in complexity.